src/types.ts

```diff
@@ -214,7 +269,15 @@ export interface PostHogConfig {
     feature_flag_request_timeout_ms: number
     get_device_id: (uuid: string) => string
     name: string
+    /**
+     * this is a read-only function that can be used to react to event capture
+     */
     _onCapture: (eventName: string, eventData: CaptureResult) => void
+    /**
+     * This function - if provided - is called immediately before sending data to the server.
+     * It allows you to edit data before it is sent, or choose not to send it all.
+     */
+    beforeSend: (cr: CaptureResult) => CaptureResult | null
```

Comment:
Rather `before_send` or? I know `_onCapture` is not following the pattern but that feels like a mistake.